### ISSUE TITLE
Support typing in cls argument of parse

### ISSUE THREAD

**Issue by mike0sv** [antidot/Pyckson#9]
Sometimes you want to parse a list of objects, for example. It would be useful to specialize it via passing typing.List[<cls>] as cls argument to .parse. Here is the minimal example.

```import json
from typing import List

import pyckson


class A:
    def __init__(self, name: str):
        self.name = name


if __name__ == '__main__':
    data = [{"name": "test"}]
    payload = json.dumps(data)
    result = pyckson.loads(List[A], payload)
```

**Comment by JGiard** [comment]
That seems like a good suggestion, I will look into it.

**Comment by JGiard** [comment]
should be fixed in v1.6
